lib/Twig/Extensions/Extension/Text.php

```diff
@@ -55,24 +55,43 @@ function twig_truncate_filter(Twig_Environment $env, $value, $length = 30, $pres
 
     function twig_wordwrap_filter(Twig_Environment $env, $value, $length = 80, $separator = "\n", $preserve = false)
     {
-        $sentences = array();
-
         $previous = mb_regex_encoding();
         mb_regex_encoding($env->getCharset());
 
-        $pieces = mb_split($separator, $value);
-        mb_regex_encoding($previous);
+        $lines = mb_split($separator, $value);
 
-        foreach ($pieces as $piece) {
-            while (!$preserve && mb_strlen($piece, $env->getCharset()) > $length) {
-                $sentences[] = mb_substr($piece, 0, $length, $env->getCharset());
-                $piece = mb_substr($piece, $length, 2048, $env->getCharset());
+        foreach ($lines as &$line) {
+            $line = rtrim($line);
+            if (mb_strlen($line, $env->getCharset()) <= $length) {
+                continue;
             }
 
-            $sentences[] = $piece;
+            $words = explode(' ', $line);
```

Comment:
you have a reference mismatch here, as you pass the reference variable to a function not taking a reference. Thus, you don't need the reference at all. You could just keep building the ``$sentences`` array instead. Please switch back to this architecture.

Btw, dealing with references makes code harder to understand, as most PHP devs are not familiar with them and with edges cases related to them (were you aware of the performance impact of reference mismatches in PHP 5.x ?)